src/lib/Server/Output/ValueObjectVisitor/Location.php

```diff
@@ -174,4 +184,31 @@ protected function visitLocationAttributes(Visitor $visitor, Generator $generato
 
         $generator->endObjectElement('ContentInfo');
     }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    private function resolveMainLocation(
+        Content\ContentInfo $contentInfo,
+        Content\Location $location
+    ): ?Content\Location {
+        $mainLocationId = $contentInfo->mainLocationId;
```

Comment:
Avoid magic due to performance please. This method should be available in 3.3 already:
```suggestion
        $mainLocationId = $contentInfo->getMainLocation();
```

Reply:
Hmm, we want to fetch `mainLocationId` though, not main Location object.

Reply:
> Hmm, we want to fetch `mainLocationId` though, not main Location object.

Ok, without ezplatform-kernel improvement providing strict getter, this doesn't make sense.